server/gitlab/api.go

```diff
@@ -412,3 +412,35 @@ func (g *gitlab) ResolveNamespaceAndProject(
 	}
 	return "", "", ErrNotFound
 }
+
+func (g *gitlab) TriggerNewBuildPipeline(user *UserInfo, repo interface{}, refToCommit *string) (*internGitlab.Pipeline, error) {
+	var err error
+	client, err := g.gitlabConnect(*user.Token)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		pipeline       *internGitlab.Pipeline
+		ctx, ctxCancel = context.WithTimeout(context.Background(), DefaultRequestTimeout)
+	)
+
+	defer ctxCancel()
+	errGroup, _ := errgroup.WithContext(ctx)
+	errGroup.Go(func() error {
+		p, response, gitError := client.Pipelines.CreatePipeline(repo, &internGitlab.CreatePipelineOptions{
```

Comment:
Will this block until the pipeline is done?

Reply:
yes the code in lib looks blocking in nature.